pkg/server/delta/v3/server.go

```diff
@@ -170,33 +171,17 @@ func (s *server) processDelta(str stream.DeltaStream, reqCh <-chan *discovery.De
 				s.subscribe(req.GetResourceNamesSubscribe(), watch.state.ResourceVersions)
 				s.unsubscribe(req.GetResourceNamesUnsubscribe(), watch.state.ResourceVersions)
 
-				watch.responses, watch.cancel = s.cache.CreateDeltaWatch(req, watch.state)
-
-				// Go does not allow for selecting over a dynamic set of channels
-				// so we introduce a termination chan to handle cancelling any watches.
-				terminate := make(chan struct{})
-				watch.termination = terminate
+				watch.cancel = s.cache.CreateDeltaWatch(req, watch.state, watch.responses)
 				watches.deltaWatches[typeURL] = watch
+
 				go func() {
-					select {
-					case resp, more := <-watch.responses:
-						if more {
-							watches.deltaMuxedResponses <- resp
-						} else {
-							// Check again if the watch is cancelled.
-							select {
-							case <-terminate: // do nothing
-							default:
-								// We cannot close the responses channel since it can be closed twice.
-								// Instead we send a fake error response.
-								watches.deltaMuxedResponses <- deltaErrorResponse
-							}
-						}
-						break
-					case <-terminate:
-						break
+					resp, more := <-watch.responses
+					if more {
+						watches.deltaMuxedResponses <- resp
 					}
 				}()
+			} else {
+				watch.Cancel()
```

Comment:
Is this new behavior?

Reply:
Ah yes I thought I removed that. It was to address: https://github.com/envoyproxy/go-control-plane/issues/446 but it seems https://github.com/envoyproxy/go-control-plane/issues/448 has outlined some more work to be done. I can remove this since it's out of scope for this PR.